Feature/hints viewer

radurentea opened the pull request:
## Description

Added Hints Viewer. Hints are provided in yaml file from esp-idf folder: <ESP-IDF FOLDER>/tools/idf_py_actions/hints.yml

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output

1. Open ERROR HINTS bottom panel:
![image](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/9642fab8-2a49-4881-9632-105a9bc44130)

2. Press on "ESP-IDF: Search Error Hint" 
![image](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/200859cf-47fe-443c-a3a1-499c38a16a6a)

3. Paste error in the top input that shows up
![image](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/be830e57-8654-4695-bb32-b7452ff20cb8)

- Expected behaviour:
Hints shown in the panel
![image](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/104757d5-9128-48d0-bb22-8d588720847d)


## How has this been tested?

Followed the steps above on a windows OS.

**Test Configuration**:
* ESP-IDF Version: 5.1.1
* OS (Windows,Linux and macOS): Windows

## Checklist
- [ ] PR Self Reviewed
- [x] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

brianignacio5 commented on src/espIdf/hints/index.ts:
```diff
@@ -0,0 +1,153 @@
+      workspace
+    ) as string;
+    const hintsPath = getHintsYmlPath(espIdfPath);
+    const fileContents = fs.readFileSync(hintsPath, "utf-8");
```
Maybe we can add some file exist validation here ? It could fail if the file doesn't exists. Also which versions of ESP-IDF support hints ? Maybe a version validation would be good to know.

Could we replace readFileSync with a Promise or async version like the one from fs-extra ?

brianignacio5 reviewed (changes requested):
Few changes I think is need to make sure the code doesn't break like file exists and IDF version validation.

AndriiFilippov commented:
@radurentea hi ! 

Tested: 
OS - Windows 10 
ESP-IDF: v5.1.1

1) Create a "Hello World" project -> add string [#import "spiram.h"] -> build -> get error "spiram.h: No such file or directory" -> call "ESP-IDF: Search Error Hint" -> paste error message -> could not find error hint: 
![image](https://github.com/espressif/vscode-esp-idf-extension/assets/69584713/faced572-42ce-437e-8391-53a0595db93e)



2) I don't understand the purpose of command "ESP-IDF: Focus on Error Hints View". I mean the Error Hint View opens automatically once user use command  "ESP-IDF: Search Error Hint". So, I don't know if this is useful feature.

radurentea commented:
Hi @AndriiFilippov,

1) Fixed the searching issue. Also, fixed an issue where if you would search directly from the Command Palette the Hints Panel wouldn't focus. (For example, if in your bottom panel you had Terminal opened).

 2) That command is added automatically by VS Code for new views. I can't change it's behaviour

AndriiFilippov commented:
@radurentea hi ! 

Create a "Hello World" project -> add string [#import "spiram.h"] -> build -> get error "spiram.h: No such file or directory" -> call "ESP-IDF: Search Error Hint" -> paste error message -> able to see hints 
![image](https://github.com/espressif/vscode-esp-idf-extension/assets/69584713/ce05a490-72f1-4244-aafa-24ba533e50af)

command "ESP-IDF: Focus on Error Hints View" works fine 👍 

LGTM 👍

brianignacio5 commented on src/espIdf/hints/index.ts:
```diff
@@ -0,0 +1,230 @@
+    ) as string;
+    const version = await utils.getEspIdfFromCMake(espIdfPath);
+
+    if (utils.compareVersion(version.trim(), "5.0") === -1) {
```
Shouldn't be ESP-IDF v5.0 or higher ? Or is only supported in v5.0 ?

radurentea replied:
It doesn't verify if it's 5.0, it actually verifies if it's higher or not than 5.0

radurentea commented:
Closing this PR because I've re-implemented this here: https://github.com/espressif/vscode-esp-idf-extension/pull/1225

The reason is that the branch is too old and there are too many conflicts to solve.
